common/src/main/java/org/alfresco/hxi_connector/common/config/properties/Retry.java

```diff
@@ -36,41 +38,56 @@
 import com.fasterxml.jackson.core.io.JsonEOFException;
 import com.fasterxml.jackson.databind.exc.MismatchedInputException;
 import lombok.AccessLevel;
-import lombok.AllArgsConstructor;
 import lombok.Data;
 import lombok.Getter;
-import lombok.NoArgsConstructor;
 import org.apache.hc.client5.http.HttpHostConnectException;
 import org.apache.hc.core5.http.MalformedChunkCodingException;
 import org.apache.hc.core5.http.NoHttpResponseException;
 
 import org.alfresco.hxi_connector.common.exception.EndpointServerErrorException;
 
 @Data
-@NoArgsConstructor
-@AllArgsConstructor
 @Getter(AccessLevel.NONE)
 public class Retry
 {
     private static final int RETRY_ATTEMPTS_DEFAULT = 10;
     private static final int RETRY_INITIAL_DELAY_DEFAULT = 500;
     private static final double RETRY_DELAY_MULTIPLIER_DEFAULT = 2;
-    private static final Set<Class<? extends Throwable>> RETRY_REASONS = Set.of(
+    private static final Set<Class<? extends Throwable>> RETRY_REASONS_BASIC = Set.of(
             EndpointServerErrorException.class,
             UnknownHostException.class,
             MalformedURLException.class,
             JsonEOFException.class,
-            MismatchedInputException.class,
-            HttpHostConnectException.class,
-            NoHttpResponseException.class,
-            MalformedChunkCodingException.class);
+            MismatchedInputException.class);
 
     @Min(-1)
     private int attempts = RETRY_ATTEMPTS_DEFAULT;
     @PositiveOrZero
     private int initialDelay = RETRY_INITIAL_DELAY_DEFAULT;
     @Positive private double delayMultiplier = RETRY_DELAY_MULTIPLIER_DEFAULT;
-    @NotNull private Set<Class<? extends Throwable>> reasons = RETRY_REASONS;
+    @NotNull private Set<Class<? extends Throwable>> reasons;
+
+    public Retry()
+    {
+        this.reasons = Stream.concat(RETRY_REASONS_BASIC.stream(), Stream.of(HttpHostConnectException.class, NoHttpResponseException.class, MalformedChunkCodingException.class))
+                .collect(Collectors.toSet());
+    }
+
+    public Retry(int attempts, int initialDelay, double delayMultiplier)
+    {
+        this();
```

Comment:
You could change the order of the constructors so that the default constructor calls the 3-arg and the 3-arg calls the 4-arg.

Reply:
Good hint! I will fix it